packages/typegpu/src/core/root/rootTypes.ts

```diff
@@ -56,11 +57,16 @@ export interface WithCompute {
   createPipeline(): TgpuComputePipeline;
 }
 
+type UndecorateRecord<T extends Record<string, unknown>> = {
+  [Key in keyof T]: Undecorate<T[Key]>;
+};
```

Comment:
Could we colocate the `UndecorateRecord` utility type alongside `Undecorate`, in the `attributes.ts` file?